apps/dh/api-dh/source/DataHub.WebApi/Modules/Charges/Client/ChargesClient.cs

```diff
@@ -89,22 +87,15 @@ public async Task<IEnumerable<Charge>> GetChargesByTypeAsync(
        ChargeType type,
        CancellationToken ct = default)
     {
-        var currentUser = httpContext.CreateUserIdentity();
-        var ownerGln = currentUser.ActorNumber.Value;
+        var user = httpContext?.HttpContext?.User;
 
-        if (currentUser.ActorRole == ActorRole.SystemOperator || currentUser.ActorRole == ActorRole.EnergySupplier)
-        {
-            ownerGln = (await marketParticipantClient_V1.ActorGetAsync(ct))
-            .Where(x => x.MarketRole.EicFunction == DataHub.WebApi.Clients.MarketParticipant.v1.EicFunction.SystemOperator).SingleOrDefault()?.ActorNumber.Value;
-        }
-
-        if (string.IsNullOrWhiteSpace(ownerGln))
+        if (user == null)
```

Comment:
question: What is the difference between these two approaches? I'm using `CreateUserIdentity` in my PR, so just wondering if I should use `user.GetMarketParticipantNumber` instead

Reply:
CreateUserIdentify uses something from ProcessManager, GetMarketParticipantNumber extracts it from the token. But I think it boils down to the same